evap/staff/views.py

```diff
@@ -2120,6 +2121,52 @@ class ProgramIndexView(SuccessMessageMixin, SaveValidFormMixin, FormsetView):
     success_message = gettext_lazy("Successfully updated the programs.")
 
 
+@manager_required
+def program_merge_selection(request):
+    form = ProgramMergeSelectionForm(request.POST or None)
+
+    if form.is_valid():
+        main_instance = form.cleaned_data["main_instance"]
+        other_instance = form.cleaned_data["other_instance"]
+        return redirect("staff:program_merge", main_instance.id, other_instance.id)
+
+    return render(request, "staff_program_merge_selection.html", {"form": form})
+
+
+@manager_required
+def program_merge(request, main_id, other_id):
+    main_instance = get_object_or_404(Program, id=main_id)
+    other_instance = get_object_or_404(Program, id=other_id)
+
+    if request.method == "POST":
```

Comment:
Why are we not using the form here? Although the other view prevents me from navigating to, for example, /staff/programs/1/merge/1, I can still POST to it and thereby delete the master program. Although this is only managers, and only by modifying the URL, this seems like a bug to me (the same happens with the course type merge)

Reply:
We agreed on adding an assert here that `main_instance != other_instance`